lib/Graph/Graph.cpp

```diff
@@ -2334,70 +2339,74 @@ static bool isGraphStorageNode(Node *n, const Function *F) {
   return false;
 }
 
-void Function::verify() const {
-  std::unordered_map<std::string, const Node *> NameToNode;
-
-  for (auto *V : getParent()->getConstants()) {
-    if (NameToNode.insert({V->getName(), V}).second)
-      continue;
+/// Check if \p node is uniquely identified by its name in \p nameToNode.
+/// \returns True if \p nameToNode does not map any node with
+///          \p node.getName(). False otherwise.
+/// When true is returned \p nameToNode maps \p node.getName() to \p node.
+static bool isUnique(std::unordered_map<std::string, const Node *> &nameToNode,
```

Comment:
Currently the uniqueness is determined by the result of the insertion.
I could split the functionality but that would be slightly less efficient.
Alternatively I can use a different name for the method. Ideas?

Reply:
instead of insert we could do find in here and insert based on the result of isUnique (as you suggested).
It's less performance efficient but it's not a big perf hit.

Reply:
That or I could rename it insertAndReport.
Don’t have a strong opinion either way.

Reply:
`insertAndReport` works

Reply:
Done.